src/libexpr/parser.y

```diff
@@ -577,7 +581,10 @@ Expr * EvalState::parseExprFromFile(const Path & path)
 
 Expr * EvalState::parseExprFromFile(const Path & path, StaticEnv & staticEnv)
 {
-    return parse(readFile(path).c_str(), path, dirOf(path), staticEnv);
+    if (path == "/dev/stdin")
```

Comment:
This is already handled by `nix-build -`, so we don't need to special-case this, see `EvalState::parseStdin`.

Reply:
wow, nice find
```nix build``` is less usable anyway, it does not print the results (https://github.com/NixOS/nix/pull/2622)

Reply:
https://github.com/NixOS/nix/pull/2583 does this for all commands that take the `-f` flag.